mempool/clist_mempool_test.go

```diff
@@ -170,23 +170,37 @@ func TestMempoolFilters(t *testing.T) {
 		{10, PreCheckAminoMaxBytes(22), PostCheckMaxGas(0), 0},
 	}
 	for tcIndex, tt := range tests {
-		mempool.Update(1, emptyTxArr, tt.preFilter, tt.postFilter)
+		mempool.Update(1, emptyTxArr, abciResponses(len(emptyTxArr), abci.CodeTypeOK), tt.preFilter, tt.postFilter)
 		checkTxs(t, mempool, tt.numTxsToCreate, UnknownPeerID)
 		require.Equal(t, tt.expectedNumTxs, mempool.Size(), "mempool had the incorrect size, on test case %d", tcIndex)
 		mempool.Flush()
 	}
 }
 
-func TestMempoolUpdateAddsTxsToCache(t *testing.T) {
+func TestMempoolUpdate(t *testing.T) {
 	app := kvstore.NewKVStoreApplication()
 	cc := proxy.NewLocalClientCreator(app)
 	mempool, cleanup := newMempoolWithApp(cc)
 	defer cleanup()
-	mempool.Update(1, []types.Tx{[]byte{0x01}}, nil, nil)
+
+	// 1. Adds valid txs to the cache
+	mempool.Update(1, []types.Tx{[]byte{0x01}}, abciResponses(1, abci.CodeTypeOK), nil, nil)
 	err := mempool.CheckTx([]byte{0x01}, nil)
 	if assert.Error(t, err) {
 		assert.Equal(t, ErrTxInCache, err)
 	}
+
+	// 2. Removes valid txs from the mempool
+	err = mempool.CheckTx([]byte{0x02}, nil)
+	require.NoError(t, err)
+	mempool.Update(1, []types.Tx{[]byte{0x02}}, abciResponses(1, abci.CodeTypeOK), nil, nil)
+	assert.Zero(t, mempool.Size())
+
+	// 3. Removes invalid transactions from the cache, but leaves them in the mempool (if present)
+	err = mempool.CheckTx([]byte{0x03}, nil)
+	require.NoError(t, err)
+	mempool.Update(1, []types.Tx{[]byte{0x03}}, abciResponses(1, 1), nil, nil)
```

Comment:
Wouldn't this be more consistent: 
```suggestion
	mempool.Update(1, []types.Tx{[]byte{0x03}}, abciResponses(1, abci.CodeTypeEncodingError), nil, nil)
```

Reply:
Oh, I see that is a different package. NVM then.